Eslint

brettz9 opened the pull request:
chore: update eslint

Changes to the now required flat config. Diff noise due to upgrades in prettier (and a few changes in `index.test.ts` due to typescript-eslint)

Also:
- change from deprecated husky command 

(See https://github.com/typicode/husky/releases/tag/v9.0.1 )

brettz9 commented:
This one is next up in line, I think, when you may be free... We're almost there! :-)

brettz9 commented:
Hi @Gregoor ... Anything I could do to make the PR easier for review?

Gregoor commented:
Hey sorry, life kept me busy. Yeah I think I'm punting on this because I don't at all like the direction prettier has taken on formatting those ternaries. I'm watching https://github.com/prettier/prettier/issues/7940 to see if it is by design or a change is planned.

To unblock this from that I'd ask you to revert updating prettier and keep this limited to ESLint, then I'll try to give it a swift review. Is that ok?

brettz9 commented:
Sure, I've reverted the `prettier` updates and if you're amenable to it, I also added a `.ncurc.cjs` file to prevent `npm-check-updates` from finding updates for `prettier`. (With `npm-check-updates`, one can just run `ncu -u` locally to update all dependencies besides those excluded in the `.ncurc.cjs` file.)

Gregoor commented:
Sounds like a good idea to have something like this, it looks like sth similar [was added to pnpm](https://github.com/pnpm/pnpm/pull/5408), shall we try that first?

Gregoor commented on src/index.test.ts:
```diff
@@ -447,7 +447,7 @@ test.each([
   expect(zerialize(dezerialize(shape) as any)).toEqual(zerialize(schema));
   const parsed = zodexSchema.safeParse(shape);
-  expect(parsed.success).to.be.true;
+  expect(parsed.success).to.equal(true);
```
nit: I'm not against (nor for) it, but I am curious about the motivation

brettz9 replied:
Was added as a requirement by typescript-eslint...

brettz9 commented:
> Sounds like a good idea to have something like this, it looks like sth similar [was added to pnpm](https://github.com/pnpm/pnpm/pull/5408), shall we try that first?

Sure, updated to support... As per https://pnpm.io/package_json#pnpmupdateconfig , just use `pnpm update --latest`.
